src/Stratis.Bitcoin/Utilities/AverageCalculator.cs

```diff
@@ -17,6 +17,14 @@ public class AverageCalculator
         /// <summary>Samples used in calculation of the average value.</summary>
         private CircularArray<double> samples;
 
+        /// <summary>
+        /// Gets a value indicating whether this instance is empty.
+        /// </summary>
+        /// <value>
+        ///   <c>true</c> if this instance is empty; otherwise, <c>false</c>.
+        /// </value>
+        public bool IsEmpty { get { return this.samples.Count == 0; } }
```

Comment:
No need for that- blocks can't be smaller than header size so in GetAverageBlockSizeBytes you can check average to be less or equal to 80. 
Also this change breaks 2 BP tests

Reply:
moving the bugfix logic into ProcessDownloadQueueLocked method